lib/cn/cn_tree.c

```diff
@@ -1828,13 +2056,13 @@ cn_comp_compact(struct cn_compaction_work *w)
      * and kv-compaction. */
     w->cw_keep_vblks = kcompact;
 
-    err = kcompact ? cn_kcompact(w) : cn_spill(w);
+    err = kcompact ? cn_kcompact(w) : (split ? cn_split(w) : cn_spill(w));
```

Comment:
nested ternary statements are not easy to read. Can you use something else?

Reply:
hmm.. I can change this to a switch statement.

Reply:
Please do, it'll make adding new actions easier, should we need to...